expression/integration_test.go

```diff
@@ -7523,3 +7523,22 @@ func TestCastRealAsTime(t *testing.T) {
 		"<nil> <nil> <nil>",
 		"<nil> <nil> <nil>"))
 }
+
+func TestCastJSONTimeDuration(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("create table t(i INT, j JSON)")
+
+	// DATE/DATETIME/TIME will be automatically converted to json date/datetime/duration
+	tk.MustExec("insert into t values (0, DATE('1998-06-13'))")
+	tk.MustExec("insert into t values (1, CAST('1998-06-13 12:12:12' as DATETIME))")
+	tk.MustExec("insert into t values (2, DATE('1596-03-31'))")
+	tk.MustExec("insert into t values (3, CAST('1596-03-31 12:12:12' as DATETIME))")
+	tk.MustQuery("select i, cast(j as date), cast(j as datetime), cast(j as time) from t").Check(testkit.Rows(
+		"0 1998-06-13 1998-06-13 00:00:00 00:00:00",
+		"1 1998-06-13 1998-06-13 12:12:12 12:12:12",
+		"2 1596-03-31 1596-03-31 00:00:00 00:00:00",
+		"3 1596-03-31 1596-03-31 12:12:12 12:12:12",
+	))
+}
```

Comment:
can you please also add a test like `select json_type(j) from t`?

Reply:
Yes!